osu.Game/Screens/Multi/Lounge/Components/RoomsContainer.cs

```diff
@@ -115,7 +115,9 @@ private void addRooms(IEnumerable<Room> rooms)
                     {
                         if (room == selectedRoom.Value)
                         {
-                            joinSelected();
+                            if (roomManager.CanJoinRoom(room))
+                                joinSelected();
```

Comment:
I'm not convinced by the way this is done at all. This logic is all very far from `RoomManager`, wherein the request is being fired - it should be in `JoinRoom()` in my opinion.

There are a few ways to implement that, and I think that `CanJoinRoom` doesn't even maybe need to exist - can probably just override `RealtimeRoomManager`'s impl, like so:

```csharp
public override void JoinRoom(Room room, Action<Room> onSuccess = null, Action<string> onError = null)
{
    if (room.Status.Value is RoomStatusEnded)
    {
        onError?.Invoke("Cannot join room as it has already ended.");
        return;
    }

    base.JoinRoom(room, r => joinMultiplayerRoom(r, onSuccess, onError), onError);
}
```

Reply:
Ah, was thinking `onError` is not the right way for this as this case is more of guaranteed while other cases of invoking should not be possible to happen, will change back...